tests/test/scheduler/test_mpi_message_buffer.cpp

```diff
@@ -0,0 +1,119 @@
+#include <catch.hpp>
+
+#include <faabric/mpi/mpi.h>
+#include <faabric/scheduler/MpiMessageBuffer.h>
+#include <faabric/util/gids.h>
+
+using namespace faabric::scheduler;
+
+MpiMessageBuffer::Arguments genRandomArguments(bool nullMsg = true,
+                                               int overrideRequestId = -1)
+{
+    int requestId;
+    if (overrideRequestId != -1) {
+        requestId = overrideRequestId;
+    } else {
+        requestId = static_cast<int>(faabric::util::generateGid());
+    }
+
+    MpiMessageBuffer::Arguments args = {
+        requestId, nullptr, 0, 1,
+        nullptr,   MPI_INT, 0, faabric::MPIMessage::NORMAL
+    };
```

Comment:
This is a good example of why this sort of struct initialisation is not great, it's hard to tell what each of these numbers and `nullptr`s refer to.